### PR TITLE
feat(protocol): Bridge proof should only use storage proof

### PR DESCRIPTION
**Pull request by cyberhorsey**
Not sure if this is the correct way to do this, but I'd take a stab at it regardless. This is a large gas optimization for the bridge.
@dantaik because we sync the storageRoot, we can remove the account proof, and use the synced signal root instead. I'm not sure if I did it correctly, tests pass though. Please let me know if this is the appropriate way or if I missed some additional verification necessary, or removed too much.

`verifyWithAccountProof` is for `isMessageFailed`, which still requires an account proof, since it uses the Bridge's storage tree not the SignalService.

### CODE REVIEW

**Comment by dantaik** [comment]
@cyberhorsey what's the gas cost difference?

File: packages/protocol/contracts/libs/LibTrieProof.sol [hunk 1]
```diff
@@ -27,18 +27,38 @@ library LibTrieProof {
...
+            LibRLPWriter.writeBytes32(value),
+            storageProof,
+            storageRoot
+        );
+    }
+
+    /**
+     * Verifies that the value of a slot in the storage of an account is value.
+     *
+     * @param stateRoot The merkle root of state tree..
```
Comment (davidtaikocha) [inline packages/protocol/contracts/libs/LibTrieProof.sol:55]:
```suggestion
     * @param stateRoot The merkle root of state tree.
```

**Review by davidtaikocha — APPROVED** [review]

**Review by Brechtpd — APPROVED** [review]

**Comment by Brechtpd** [comment]
Just a general question, it seems like it's possible to move `isSignalReceived` out of `SignalService` pretty easily by for example making it an external library function instead. That way the the contract gets much simpler and the contract is closer to just being a storage contract with a method to store some data and a method to read some data, so less risky and then there will never be a need to update it. Now with the built in inclusion of the MPT check, the contract gets outdated when for example Ethereum changes something its trie format or a precompile is made available that is more efficient becomes available.

Also a specific reason signals cannot be `0`? Maybe a potential issue with MPT check that always returns true even when the signal is set to `0` because it's the default value? (depends on the MPT verification logic how that is handled).

**Comment by cyberhorsey** [comment]
> Just a general question, it seems like it's possible to move `isSignalReceived` out of `SignalService` pretty easily by for example making it an external library function instead. That way the the contract gets much simpler and the contract is closer to just being a storage contract with a method to store some data and a method to read some data, so less risky and then there will never be a need to update it. Now with the built in inclusion of the MPT check, the contract gets outdated when for example Ethereum changes something its trie format or a precompile is made available that is more efficient becomes available.
> 
> Also a specific reason signals cannot be `0`? Maybe a potential issue with MPT check that always returns true even when the signal is set to `0` because it's the default value? (depends on the MPT verification logic how that is handled).

I can move that logic out of SignalService to an external library method in a separate PR!

And yes, I think that is the reason it cannot be 0.

**Pull request merged**